ext-dynamic-commitments.md

```diff
@@ -0,0 +1,1086 @@
+Lightning Network. The sooner that network participants can upgrade to STCs the
+more we will have the necessary network infrastructure to be able to make
+effective use of PTLCs when the protocols for them are ready for deployment.
+
+Due to the design of STCs, and the fact that they take full advantage of the
+capabilities afforded by Schnorr Signatures, there is no way to construct a
+valid `channel_announcement` message that references the output corresponding to
+the nodes' joint public key. As such, even if we were to directly spend an
+existing funding output to a new STC funding output, and even with the provision
+in BOLT 7 to delay graph pruning by 12 blocks after the channel point is spent,
+we have no way of advertising the STC to the network at the time of writing of
+this proposal.
+
+That said, there is a development effort, concurrent with this, for a new gossip
+system that is capable of understanding the announcements of new STCs. However,
+even with a new gossip system capable of understanding the STC construction and
+announcement, it will take quite some time for such a system to be broadly
+deployed across the Lightning Network. In the interim, to combat the
+disincentive of upgrading to STCs, this proposal to enable the change of these
+channel parameters (including channel types) without requiring channel turnover
+is submitted.
+
+## Preliminaries
+
+This proposal includes a detailed section on the preliminaries to document some
+of the rationale for the design that is presented later. If you are a Bitcoin
+and Lightning Network protocol expert or you are uninterested in the thought
+process behind what is presented here, you may wish to skip to the Design
+Overview section to save time.
+
+### Channel Opening Parameters
+
+As described in BOLT 2, during the channel opening procedure there are a number
+of parameters specified in the `open_channel` and `accept_channel` messages that
+remain static over the lifetime of the channel. A subset of these are updatable
+using other messages defined in the protocol. However, after accounting for the
+channel parameters that can be changed using existing mechanisms, there remains
+a list of parameters which are unchangeable. We list these below:
+
+- dust_limit_satoshis
+- max_htlc_value_in_flight_msat
+- channel_reserve_satoshis
+- to_self_delay
+- max_accepted_htlcs
+- funding_pubkey
+- revocation_basepoint
+- payment_basepoint
+- delayed_payment_basepoint
+- htlc_basepoint
+- first_per_commitment_point
+- channel_flags
+- upfront_shutdown
+- channel_type
+
+After some analysis during the development of this proposal, we determined that
+the basepoint values don't make sense to rotate. There is no obvious value in
+doing so and it carries additional administrative costs to rotate them. Finally,
+changing the upfront_shutdown script over the lifetime of the channel is
+self-defeating and so we exclude it as well. The list of channel parameters
```

Comment:
IIUC the purpose of upfront shutdown is to prevent it from being set at a later time, allowing ourselves to set it during a dyncomm negotiation would defeat the purpose of the parameter, right?